advanced/docker/Dockerfile

```diff
@@ -0,0 +1,32 @@
+# This is the base image for the example_docker advanced example cloud execution
+FROM ubuntu:22.04
+
+# Configure apt
+ARG DEBIAN_FRONTEND=noninteractive
+RUN apt update -y && apt install -y software-properties-common
+RUN add-apt-repository -y ppa:deadsnakes/ppa
+
+# Install python3.9
+RUN apt update -y && apt install -y python3.9
+RUN rm /usr/bin/python3
+RUN ln -s /usr/bin/python3.9 /usr/bin/python3
+RUN ln -s /usr/bin/python3.9 /usr/bin/python
+
+# Install pip3
+RUN apt update -y && apt install -y python3-pip
+RUN apt update -y && apt install --reinstall -y python3.9-distutils
+
+# Set the launch command
+ENV PATH="/sematic/bin/:${PATH}"
+COPY advanced/scripts/launch.sh /advanced/scripts/launch.sh
+ENTRYPOINT ["/advanced/scripts/launch.sh"]
```

Comment:
This is *really* unfortunate. I feel like we have to find some solution such that the user doesn't have to do this. Here's a proposal:
1. We make `CONTAINER_ENTRYPOINT` a new field in the resolutions table
2. We allow passing in a `CONTAINER_ENTRYPOINT` argument from the build plugin via env var, the same way we allow the build plugin to pass in the image URI via env var
3. If that field is `null`, we don't do anything with the entrypoint when starting the container. If it is a list of strings, we use that as the container's entrypoint. We can have the `docker` one pass in `CONTAINER_ENTRYPOINT="python3 -m sematic.resolvers.worker"`

Can be done in a later PR, but I feel it really will be a big deal for user experience. Explaining to them why this is necessary would be a big pain, and having each user hard-code it will also limit our ability to do things like rename `worker.py` later & other such refactors.

Reply:
Or another option: we provide base images (with different tech stacks or OS), similar to what CircleCI offers. I know it's a lot of work (we can start with a few). But it seems to be something a mature product does.

In the base image, we could have a starter script (docker-entrypoint.sh) and `/docker-entrypoint.init.d/` for custom extensions to run.

Reply:
Good idea! Wrote it down to implement it in a subsequent pair of PRs.
Will need to test if passing arguments to the container this way will still work. I discovered that the `Dockerfile` syntax is fickle; there are many ways of specifying an entry point, but this is the only incantation that worked with arguments.

Reply:
@chance-an We provide base images on an alternate flow:
- when specifying `base_uri` in the config file (not implemented yet), that image is used as the base image, that will only need the user code to be copied
- when specifying `image_script` (this use case), we use this script to build the image from scratch; it is the user's responsibility to make it work; we just need to make it easy for them

Reply:
Yes, sorry I wasn't clear earlier. What I proposed is different than either of those. In that, we provide a base image and let the user write their own image on top of the base image. It is still the `image_script` route, but the essential stuff is already taken care of, in this way, we also `make it easy for them`.

eg: 

```
FROM sematic-scaffold-python:0.28.1
```

Reply:
Yes, that is intrinsically supported. We could document very clearly what the Sematic base image does so that the users can start from it instead of from scratch in the script version.

Reply:
Yes, providing base images to people is a good idea, and is something we already do. However, sometimes people want to "bring their own" base image; we have seen this in the case of at least one of our big users who doesn't want to use our base images.